### ISSUE TITLE
available_colors catches first character of stdin

### ISSUE THREAD

**Issue by booger71** [ticki/termion#101]
Calling stdout.available_colors() before stdin.keys() causes the first character of input to be lost.  Using the keys.rs example, 

```
fn main() {
    let stdin = stdin();
    let mut stdout = stdout().into_raw_mode().unwrap();

    // Add available_color() here.
    let colors = stdout.available_colors();

    write!(stdout,
           "{}{}q to exit. Type stuff, use alt, and so on.{}",
           termion::clear::All,
           termion::cursor::Goto(1, 1),
           termion::cursor::Hide)
            .unwrap();
    stdout.flush().unwrap();

    for c in stdin.keys() {
```

The first character typed will be lost. Full source of example attached.

[main.rs.txt](https://github.com/ticki/termion/files/907905/main.rs.txt)

**Comment by IGI-111** [comment]
I can reproduce but I'm not really sure why it's happening.

In theory, `available_colors()` only reads up to a `BEL` (from the terminal answer) and this should all happen before any user input.

What's weird is in this example `colors` has a value before any key was pressed by the user, so `available_colors()` must have terminated.

Why does stdin ignore the first input then?

I would understand if this was data lost from stdin because it predates the call to `available_colors()` (it would be similar to #64 then), but this more mysterious.

**Comment by booger71** [comment]
Additional info: It will catch the first byte, of the input. Pressing F1 in this case will catch the escape and return '[11~' instead of '^[[11~'.

Thank you for the help!

**Comment by gyscos** [comment]
Indeed. Using gdb shows that the very call to `StdIn.next()` skips the first byte. So the problem apparently happens before the byte is even sent to the application, as if the terminal itself was handling the first byte differently.

**Comment by IGI-111** [comment]
>if reading the byte wasn't blocking, we wouldn't need an AsyncReader at all

Well then we might as well go the route proposed in #7 and do actual `O_NONBLOCK` operations.
